ed448-goldilocks/src/montgomery/point.rs

```diff
@@ -0,0 +1,574 @@
+use elliptic_curve::{
+    CurveGroup, Error, Group,
+    array::Array,
+    bigint::U448,
+    consts::U57,
+    group::GroupEncoding,
+    group::cofactor::CofactorGroup,
+    group::prime::PrimeGroup,
+    ops::LinearCombination,
+    point::{AffineCoordinates, NonIdentity},
+    zeroize::DefaultIsZeroes,
+};
+use hash2curve::{ExpandMsgXof, GroupDigest};
+use rand_core::TryRngCore;
+use sha3::Shake256;
+use subtle::{Choice, ConditionallySelectable, ConstantTimeEq, CtOption};
+
+use super::{
+    DEFAULT_ENCODE_TO_CURVE_SUITE, DEFAULT_HASH_TO_CURVE_SUITE, MontgomeryScalar, MontgomeryXpoint,
+    ProjectiveMontgomeryXpoint,
+};
+use crate::field::{ConstMontyType, FieldElement};
+use crate::{AffinePoint, Curve448, Curve448FieldBytes, ORDER};
+
+/// A point in Montgomery form including the y-coordinate.
+#[derive(Copy, Clone, Debug, Default, Eq)]
+pub struct MontgomeryPoint {
+    pub(super) x: FieldElement,
+    pub(super) y: FieldElement,
+}
```

Comment:
I know the coordinate names for Montgomery are confusing as hell to begin with, but I would probably expect this to match the coordinate names for `ProjectiveMontgomeryPoint`, i.e. `U` and `V`.

Reply:
Hm, I was going after the papers here, which consistently referred to X and Y when dealing in affine form and U, V and W when dealing with projective points (AFAIR). But yeah, I don't mind ignoring all that and making it a bit easier on ourselves.